app/src/main/java/com/android/ubclaunchpad/driver/UI/EditProfileActivity.java

```diff
@@ -0,0 +1,157 @@
+package com.android.ubclaunchpad.driver.UI;
+
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import com.android.ubclaunchpad.driver.R;
+import com.android.ubclaunchpad.driver.util.FirebaseUtils;
+import com.android.ubclaunchpad.driver.util.StringUtils;
+import com.android.ubclaunchpad.driver.user.UserManager;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.FirebaseUser;
+
+public class EditProfileActivity extends BaseMenuActivity {
+    private Button nameUpdateButton;
+    private Button passwordUpdateButton;
+    private Button emailUpdateButton;
+
+    private EditText nameEditText;
+    private EditText emailEditText;
+    private EditText passwordEditText;
+
+    // this method hides "edit profile" option from the menu
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.main_menu, menu);
+
+        MenuItem item = menu.findItem(R.id.action_edit_profile);
+        item.setVisible(false);
+        return true;
+    }
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_edit_profile);
+
+        nameUpdateButton = (Button) findViewById(R.id.name_update_button);
```

Comment:
You can use Butterknife instead of the findViewByIds: http://jakewharton.github.io/butterknife/
You can find examples of us using it in RegisterActivity and SessionActivity.